core/controllers/classifier_test.py

```diff
@@ -0,0 +1,134 @@
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the controllers that communicate with VM for training
+classifiers."""
+
+import hashlib
+import hmac
+import json
+import os
+
+from core.domain import classifier_services
+from core.domain import exp_services
+from core.tests import test_utils
+import feconf
+
+
+def get_training_data_from_state(state):
```

Comment:
@pranavsid98 you only need to put get_training_data_from_state() in classifier_services or exploration_services (sounds like a better place). It is a very minor change that I'm suggesting and not complete refactoring.